### ISSUE TITLE
edge case error in json forms

### ISSUE THREAD

**Issue by tghoward** [geopaparazzi/geopaparazzi#470]
it looks like this entry in a form (the first entry: completely empty brackets!):
`
"formitems": [{
    },{
        "key": "an integer number",
        "value": "",
        "type": "integer"
    }
]
`

is allowed by the jsonlint validator but is not handled well by geopaparazzi. This is definitely a bad form and the code is wrong (!) but since the validator allows it through, perhaps it should be caught by geopap?  Right now the behaviour is for Geopap to display the form and allow entry into all fields, but this completely empty field is created with a name as "-" and any attempt to save results in this message: 

"Please check that the field "-" is valid. Not validated contraints:."

So the data entered in other fields can not be saved.

**Comment by tghoward** [comment]
Hi @moovida , 
Sure, here you go. I've modified one of the 'forms' from the default tags.json. 
[tags.txt](https://github.com/geopaparazzi/geopaparazzi/files/1886317/tags.txt)

github didn't allow me to upload it as a *.json file so I changed the extension to .txt. Download, change the extension back to .json, and put in the geopaparazzi folder on your gadget. 

1. run the note, choose text note
2. see that there are three fields. The second and third fields work as expected. The first field allows any entry but the form can not be saved, no matter what is put into the first field. 

Thanks!
